src/sipp.cpp

```diff
@@ -990,7 +990,9 @@ static void help_stats()
 static void print_last_stats()
 {
     interrupt = 1;
-    sp->print_closing_stats();
+    if (sp) {
+      sp->print_closing_stats();
```

Comment:
I see the indentation check doesn't work flawlessly ;)

Reply:
```
diff --git a/validate-src.sh b/validate-src.sh
index f720ffb..80ba0cf 100755
--- a/validate-src.sh
+++ b/validate-src.sh
@@ -49,13 +49,13 @@ fi
 files_with_not_4_spaces=$(
     find include src '!' -name config.h \
       '(' -name '*.cpp' -o -name '*.hpp' -o -name '*.c' -o -name '*.h' ')' |
-    xargs grep -lE '^(  \}| {1,3}[a-z_])')
+    xargs grep -lE '^(    )* {1,3}[^ *]')
 if test -n "$files_with_not_4_spaces"; then
     echo "files with non-standard indentation found:" >&2
     for file in $files_with_not_4_spaces; do
         echo "  $file" >&2
         if $print_listing; then
-            grep -nE '^(  \}| {1,3}[a-z_])' $file |
+            grep -nE '^(    )* {1,3}[^ *]' $file |
                 sed -e "$list_sed" | cat -A >&2
         fi
     done
```

Reply:
I've actually been thinking of switching to something more hardened like clang-format (once the count of open pull requests goes down a bit, because it'll cause merge conflicts).